pkg/test/framework.go

```diff
@@ -51,10 +53,31 @@ type Framework struct {
 	DynamicClient     dynclient.Client
 	DynamicDecoder    runtime.Decoder
 	NamespacedManPath *string
+	InCluster         bool
 }
 
 func setup(kubeconfigPath, namespacedManPath *string) error {
-	kubeconfig, err := clientcmd.BuildConfigFromFlags("", *kubeconfigPath)
+	var err error
+	var kubeconfig *rest.Config
+	inCluster := false
+	if *kubeconfigPath == "incluster" {
+		// Work around https://github.com/kubernetes/kubernetes/issues/40973
+		// See https://github.com/coreos/etcd-operator/issues/731#issuecomment-283804819
```

Comment:
nit: We can remove the etcd-operator issue link since we're linking to the upstream issue in the line above.